parser/src/plonky2.rs

```diff
@@ -307,7 +308,19 @@ impl EdgeBlockTrace {
 
                 Ok(payload)
             })
-            .collect()
+            .collect::<TraceParsingResult<Vec<_>>>()?;
+
+        Ok(match tx_proof_gen_ir.len() {
+            0 => vec![
+                TxnProofGenIR::create_dummy(b_height, 0),
+                TxnProofGenIR::create_dummy(b_height, 1),
+            ],
+            1 => {
+                tx_proof_gen_ir.push(tx_proof_gen_ir[0].clone_as(b_height, 1));
+                tx_proof_gen_ir
+            }
+            _ => tx_proof_gen_ir,
```

Comment:
Almost...

The `0` case is fine, but the `1` case I think still needs to call `create_dummy`. I might have not really explained the `1` case well in Slack, but only some fields in `create_dummy` need to be copied from the previous txn. Like I think this would work:

```rust
/// Creates a dummy txn that appears right after a non-dummy txn.
/// 
/// This will only occur when a block has `1` txn inside it. In this special case, the dummy txn needs some information from the previous txn.
pub fn create_dummy_following_real_txn(prev_real_txn: &TxnProofGenIR, b_height: BlockHeight, txn_idx: TxnIdx) -> Self {
    let deltas = ProofBeforeAndAfterDeltas {
        gas_used_before: prev_real_txn.deltas.gas_used_after,
        gas_used_after: prev_real_txn.deltas.gas_used_after,
        block_bloom_before: prev_real_txn.deltas.block_bloom_after,
        block_bloom_after: prev_real_txn.deltas.block_bloom_after,
    };

    let mut dummy_txn = Self::create_dummy(b_height, txn_idx);
    dummy_txn.trie_roots_after = prev_real_txn.trie_roots_after.clone();
    dummy_txn.deltas = deltas;

    dummy_txn
}
```

I think the number of LoC is large enough that we can probably just create a separate dummy function, but let me know what you think. I'll do a separate PR to the prover lib to add this.